protocol.md

```diff
@@ -334,6 +335,34 @@ Since metadata can contain arbitrary binary values, Servers SHOULD
 carefully validate metadata values or sanitize them before using them
 as header values to avoid header smuggling.
 
+Clients and servers SHOULD implement the metadata key ``filetype``
+with the value containing the actual ``Content-Type`` of the Upload
+(because the ``Content-Type`` header is formally required to be set to
+``application/offset+octet-stream`` for [Creation with
+Upload](#creation-with-upload)).
+
+Clients and servers MAY support additional well known metadata keys:
+
+* ``filename`` for a common file name
+
+The specific metadata keys documented herein are reserved for the
+respective use and MUST NOT be used for other purposes.
```

Comment:
I agree with @smatsson that this sentence is too restrictive. It would be a breaking change and there are no plans to make a major release for tus. I don't think this would serve the ecosystem well.

That being said, I am happy to recommend (not force) the usage of the filename and filetype metadata keys for the purposes as laid out here.